src/mercury_engine_data_structures/formats/bmssd.py

```diff
@@ -13,133 +17,245 @@
+    "lights"
+    / make_vector(  # only used in MSR
         Struct(
-            str1=StrId,
-            elems=make_vector(
-                Struct(
-                    float1=CVector3D,
-                    float2=CVector3D,
-                    float3=CVector3D,
-                )
-            ),
+            "model_name" / StrId,
+            "char2" / Byte,
+            "char3" / Byte,
+            "char4" / Byte,
+            "int5" / Int32ul,
+            "int6" / Int32ul,
+            "int7" / Int32ul,
+            "char8" / Byte,
+            "char9" / Byte,
+            "int10" / Int32ul,
+            "float13" / CVector3D,
+            "int11" / Int8ul,
         )
     ),
-    strings_a=make_vector(StrId),
-    unk_structs_a=game_check.is_at_most(
-        Game.SAMUS_RETURNS,
-        make_vector(
-            Struct(
-                str1=StrId,
-                char2=Byte,
-                char3=Byte,
-                char4=Byte,
-                int5=Int32ul,
-                int6=Int32ul,
-                int7=Int32ul,
-                char8=Byte,
-                char9=Byte,
-                int10=Int32ul,
-                float13=CVector3D,
-                int11=Int8ul,
-            )
-        ),
-        make_vector(
-            Struct(
-                str1=StrId,
-                char2=Byte,
-                char3=Byte,
-                char4=Byte,
-                int5=Int32ul,
-                int6=Int32ul,
-                int7=Int32ul,
-                char8=Byte,
-                char9=Byte,
-                int10=Int32ul,
-                str11=StaticPaddedString(16, "utf-8"),
-                int12=Int32ul,
-                float13=CVector3D,
-                float14=CVector3D,
-                float15=CVector3D,
-                int16=Int32ul,
-                float17=CVector3D,
-            )
-        ),
-    ),
-    strings_b=make_vector(
-        StrId,
+    Const(0, Int32ul),  # likely unused array
+    "scene_groups"
+    / make_vector(
+        Struct(
+            "sg_name" / StrId,
+            "item_count" / Int32ul,
+            "item_groups" / make_dict(make_vector(game_check.is_sr_or_else(Int32ul, Int64ul)), Int32ul),
+        )
     ),
-    scene_groups=game_check.is_at_most(
-        Game.SAMUS_RETURNS,
-        make_vector(
-            Struct(
-                sg_name=StrId,
-                models_per_sg=Int32ul,
-                model_groups=make_vector(
-                    Struct(
-                        model_group=Int32ul,
-                        models=make_vector(
-                            Struct(
-                                model_id=Int32ul,
-                            )
-                        ),
+    construct.Terminated,
+).compile()
+
+
+def crc_func(obj):
+    return crc32 if obj._version == "1.12.0" else crc64
+
+
+class BmssdAdapter(Adapter):
+    ItemTypes = {
+        0: "scene_blocks",
+        1: "objects",
+        2: "lights",
+    }
+
```

Comment:
surely you can use the `ItemType` enum for whatever this is doing?

Reply:
I realized just now that the way i resolved this (using the enums as dict keys) breaks dumping json through cli. should I make it use strings to prevent this?

Reply:
Could possibly make it a IntEnum (or a StrEnum which is better for dumping but requires py3.11)

Reply:
`class ItemType(str, Enum)` will do anything you may need `StrEnum` for

Reply:
about that, lmao. 

https://tomwojcik.com/posts/2023-01-02/python-311-str-enum-breaking-change

i think IntEnum would still work but eugh, StrEnum would be so much nicer.

Reply:
ok, so override `__format__()` if you need to guarantee specific behavior from it? I don't really see what the problem is